### PR TITLE
Improve support for Nix+Haskell tooling 

### PR DESCRIPTION
**Pull request by riz0id**
This PR makes minor additions to better support integration with Nix and Haskell tooling like HLS.

### CODE REVIEW

File: hie.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+cradle:
+  cabal:
+    - path: "./src"
+      component: "lib:grpc-mqtt"
+    - path: "./test"
+      component: "test:test"
```
Comment (Friede80) [inline hie.yaml:6]: Is this file necessary? It seems to work fine for me without this. HLS will create and use a default `hie.yaml` if none is present.

**Review by Friede80 — APPROVED** [review]
Awesome! Thanks for adding this!

File: hie.yaml (hunk 1 above)
Comment (riz0id) [reply hie.yaml:6]: My HLS fails during initialization with an error message complaining about something along the lines of `cabal --exec 'ghc --version'` without it for some reason. I figure it doesn't hurt to add and may become necessary once the `template-haskell` parts of `grpc-mqtt` get split into a separate package.
